sherpa/optmethods/optoptimagic.py

```diff
@@ -0,0 +1,327 @@
+#
+#  Copyright (C) 2025
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+"""Interface to optimagic optimization methods.
+
+This module contains classes that wrap the optimization functions in
+`optimagic <https://estimagic.org>`_ to match the calling signature and 
+return values to the Sherpa interface.
+
+If `optimagic <https://estimagic.org>`_ is installed, classes are created automatically and
+can be used in the same way as other optimizers in Sherpa.
+
+Optimagic supports well over 50 different algorithms
+----------------------------------------------------
+Optimagic supports several different optimizers from about a dozen different
+libraries, including scipy, nlopt, pygmo2, and others; most if these dependencies
+need to be installed separately; they will only be available to you if the
+respective library is installed in your Python environment.
+
+While optimagic aims to provide a consistent interface to all of these libraries,
+the options that can be set for those optimizers depends on the algorithm,
+e.g. in some cases one may set the maximum number of iterations, while in others
+one may define the direction of the optimization. The 
+`optimagic documentation <https://optimagic.readthedocs.io/en/latest/algorithms.html>`_
+lists all available algorithms and their options.
+
+There is no default algorithm, so you always need to specify the 
+algorithm to use before you can use the optimizer in a fit:
+
+    >>> from sherpa.optmethods import Optimagic
+    >>> opt = Optimagic(name='my_cool_optimizer')
+    >>> opt.algorithm = 'scipy_ls_dogbox'
+
+Setting additional algorithm options is supported, but not required
+(all options have defaults):
+
+    >>> opt.algo_options = {'stopping.maxfun': 1000}
+
+Most inputs for optimagic are automatically filled by Sherpa
+------------------------------------------------------------
+This module wraps the `optimagic.minimize` function in such a way that
+all the information that Sherpa already has about the model, the data, and the
+statistic is passed to the optimizer. Sherpa will automatically pick the correct
+format of the statistic function, convert the parameter minimum and maximum
+to the format for optimagic bounds etc.
```

Comment:
```suggestion
format of the statistic function and convert the parameter minimum and maximum
to the format for optimagic bounds.
```

It's not clear to me what `etc` means here, so just remove it. Unless there's something explicit that can be used instead?